tests/admin_views/tests.py

```diff
@@ -1912,6 +1912,102 @@ def test_change_view_save_as_new(self):
         new_article = Article.objects.latest('id')
         self.assertRedirects(post, reverse('admin:admin_views_article_change', args=(new_article.pk,)))
 
+    def test_change_view_with_view_only_inlines(self):
+        """
+        User with change permission to a section but view-only for inlines.
+        """
+        self.viewuser.user_permissions.add(get_perm(Section, get_permission_codename('change', Section._meta)))
+        self.client.force_login(self.viewuser)
+
+        # GET shows inlines
+        response = self.client.get(reverse('admin:admin_views_section_change', args=(self.s1.pk,)))
+        self.assertEqual(
+            len(response.context['inline_admin_formsets']), 1,
+            'Article inline FormSet should be shown on GET'
+        )
+        article_inline_formset = response.context['inline_admin_formsets'][0]
+        self.assertEqual(len(article_inline_formset.forms), 3)
+
+        # Valid POST is accepted
+        data = {
+            'name': 'Can edit name with view-only inlines',
+            'article_set-TOTAL_FORMS': 3,
+            'article_set-INITIAL_FORMS': 3
+        }
+        response = self.client.post(reverse('admin:admin_views_section_change', args=(self.s1.pk,)), data)
+        self.assertRedirects(response, reverse('admin:admin_views_section_changelist'))
+        self.assertEqual(Section.objects.get(pk=self.s1.pk).name, data['name'])
+
+        # Invalid POST reshows inlines
+        del data['name']
+        response = self.client.post(reverse('admin:admin_views_section_change', args=(self.s1.pk,)), data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            len(response.context['inline_admin_formsets']), 1,
+            'Article inline FormSet should be shown for invalid POST'
+        )
+        article_inline_formset = response.context['inline_admin_formsets'][0]
+        self.assertEqual(len(article_inline_formset.forms), 3)
+
+    def test_change_view_with_view_plus_add_inlines(self):
+        self.viewuser.user_permissions.add(get_perm(Section, get_permission_codename('change', Section._meta)))
+        self.viewuser.user_permissions.add(get_perm(Article, get_permission_codename('add', Article._meta)))
+        self.client.force_login(self.viewuser)
+
+        # GET shows inlines
+        response = self.client.get(reverse('admin:admin_views_section_change', args=(self.s1.pk,)))
+        self.assertEqual(
+            len(response.context['inline_admin_formsets']), 1,
+            'Article inline FormSet should be shown on GET'
```

Comment:
I consider this redundant with the `# GET shows inlines` and the comment is more inline with existing tests.